### PR TITLE
Add API and CLI calls for querying database dumps

### PR DESCRIPTION
**Pull request by johnelse**

### CODE REVIEW

**Comment by djs55** [comment]
Looks like quite a useful thing!

**Comment by robhoes** [comment]
"johnelse added some commits about 4 years ago" !!!

**Comment by djs55** [comment]
got a bit lost did they?

On Mon, Aug 10, 2015 at 8:14 PM, Rob Hoes <notifications@github.com> wrote:

> "johnelse added some commits about 4 years ago" !!!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xapi-project/xen-api/pull/2331#issuecomment-129570562>
> .
>



-- 
Dave Scott

**Comment by johnelse** [comment]
The world is finally ready! (Actually @germanop was asking if we had a command to do this)

File: ocaml/idl/datamodel.ml [hunk 1]
```diff
@@ -1360,6 +1360,15 @@ let slave_local_login_with_password = call ~flags:[]
   ~allowed_roles:_R_POOL_ADMIN (*only root can do an emergency slave login*)
   ()
 
+let session_create_from_db_file = call
+	~in_product_since:rel_boston
```
Comment (robhoes) [inline ocaml/idl/datamodel.ml]: Really?

Comment (johnelse) [inline ocaml/idl/datamodel.ml]: Whoops!

**Comment by robhoes** [comment]
Is this able to deal with database files of older schema versions? That is, can I use this to inspect a Clearwater DB on a Dundee host?

**Comment by johnelse** [comment]
@robhoes good question, I will try it.

**Comment by johnelse** [comment]
@robhoes it didn't work, but my latest commit fixes it :-)

**Comment by robhoes** [comment]
@johnelse Well done! Let's get this merged now, shall we...

**Pull request merged**